src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Automation/VsBuildManager.cs

```diff
@@ -0,0 +1,180 @@
+        {
+            get { return _vsProject.Project; }
+        }
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        public virtual EnvDTE.DTE DTE
+        {
+            get { return _vsProject.DTE; }
+        }
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        public virtual object DesignTimeOutputMonikers
+        {
+            get
+            {
+                var project = _unconfiguredProjectServices.ActiveConfiguredProject;
+                var ruleSource = project.Services.ProjectSubscription.ProjectRuleSource;
+                var update = _unconfiguredProjectServices.ThreadingService.ExecuteSynchronously(() => ruleSource.GetLatestVersionAsync(project, new string[] { Compile.SchemaName }));
+                var snapshot = update[Compile.SchemaName];
+
+                var monikers = new List<string>();
+                foreach (var item in snapshot.Items.Values)
+                {
+                    bool isLink = GetBooleanPropertyValue(item, Compile.LinkProperty);
+                    bool designTime = GetBooleanPropertyValue(item, Compile.DesignTimeProperty);
+
+                    if (!isLink && designTime)
+                    {
+                        if (item.TryGetValue(Compile.FullPathProperty, out string path))
+                        {
+                            monikers.Add(path);
+                        }
+                    }
+                }
+                
+                return monikers.ToArray();
+
+                bool GetBooleanPropertyValue(System.Collections.Immutable.IImmutableDictionary<string, string> item, string propertyName)
+                {
+                    return item.TryGetValue(propertyName, out string value) && StringComparers.PropertyValues.Equals(value, "true");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1065:DoNotRaiseExceptionsInUnexpectedLocations")]
+        public virtual object Parent
+        {
+            get { throw new NotImplementedException(); }
+        }
+
+        #region BuildManager Members
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        public virtual string BuildDesignTimeOutput(string bstrOutputMoniker)
+        {
+            var languageName = _unconfiguredProjectServices.ThreadingService.ExecuteSynchronously(async () =>
+            {
+                var property = await _unconfiguredProjectServices.ActiveConfiguredProjectProperties.GetConfigurationGeneralPropertiesAsync();
+                return await property.LanguageServiceName.GetValueAsync();
+            });
+
+            // TODO: Once Roslyn work is complete:
+            //
+            //   var compiler = _serviceProvider.GetService(typeof(Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.CPS.ITempPECompilerHost));
```

Comment:
Use IVsService<T>.

Reply:
As in `IVsService<ITempPECompilerHost>`? I'll try it out when I get up to that bit, but in the experimentation to get to here, roslyn exports weren't available when this was being constructed

Reply:
This isn't an export - this is a proffered service. I think if this setup correctly, Roslyn package should be initialized by the fact that you are pull on the service. IVsService is identical to this except it forces UI threadiness.